gnome-initial-setup/gis-driver.c

```diff
@@ -288,6 +294,118 @@ gis_driver_get_mode (GisDriver *driver)
 }
 
 gboolean
+gis_driver_is_in_demo_mode (GisDriver *driver)
+{
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+  return priv->is_in_demo_mode;
+}
+
+#define DEMO_ACCOUNT_USERNAME "demo-guest"
+#define DEMO_ACCOUNT_FULLNAME "Demo"
+
+static void
+handle_demo_mode_error (GError *error)
+{
+  GtkWidget *dialog;
+
+  g_warning ("%s", error->message);
+  dialog = gtk_message_dialog_new (NULL,
+                                   GTK_DIALOG_DESTROY_WITH_PARENT,
+                                   GTK_MESSAGE_ERROR,
+                                   GTK_BUTTONS_CLOSE,
+                                   "Failed to enter demo mode: %s",
+                                   error->message);
+  gtk_dialog_run (GTK_DIALOG (dialog));
+  gtk_widget_destroy (dialog);
+  g_error_free (error);
+}
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (ActUser, g_object_unref)
+
+static gboolean
+create_demo_user (GisDriver *driver)
+{
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+  g_autoptr(ActUser) user;
+  GError *error = NULL;
+  const gchar *language;
+
+  error = NULL;
+  user = act_user_manager_create_user (act_user_manager_get_default (),
+                                       DEMO_ACCOUNT_USERNAME,
+                                       DEMO_ACCOUNT_FULLNAME,
+                                       ACT_USER_ACCOUNT_TYPE_ADMINISTRATOR,
+                                       &error);
+  if (error)
+    {
+      handle_demo_mode_error (error);
+      return FALSE;
+    }
+
+  act_user_set_password_mode (user, ACT_USER_PASSWORD_MODE_NONE);
+  act_user_set_automatic_login (user, FALSE);
+
+  language = gis_driver_get_user_language (driver);
+
+  if (language)
+    act_user_set_language (user, language);
+
+  gis_driver_set_user_permissions (driver, user, NULL);
+  if (!gis_pkexec (LIBEXECDIR "/eos-setup-demo-user",
+                   "user",
+                   DEMO_ACCOUNT_USERNAME,
+                   &error))
+    {
+      handle_demo_mode_error (error);
+      return FALSE;
+    }
+
+  if (!gis_pkexec (LIBEXECDIR "/eos-setup-demo-user-autologin",
+                   NULL,
+                   NULL,
+                   &error))
+    {
+      handle_demo_mode_error (error);
+      return FALSE;
+    }
+
+  gis_update_login_keyring_password ("");
+
+  return TRUE;
+}
+
+void
+gis_driver_enter_demo_mode (GisDriver *driver)
+{
+  GisDriverPrivate *priv = gis_driver_get_instance_private (driver);
+
+  if (priv->is_in_demo_mode)
+    return;
+
+  /* Enter test mode.
+   *
+   * After this point, all data on the system will become ephemeral,
+   * including /home, which means any changes will be lost on reboot.
+   */
+  GError *error = NULL;
+  if (!gis_pkexec (LIBEXECDIR "/eos-test-mode", NULL, NULL, &error)) {
```

Comment:
Different parts of the code you added to this file use different coding style for block indentation. I think the style of the file is overall using curly braces on their own line, instead of this.